deepchecks/core/display.py

```diff
@@ -0,0 +1,368 @@
+    @abc.abstractmethod
+    def widget_serializer(self) -> WidgetSerializer[t.Any]:
+        """Return WidgetSerializer instance."""
+        raise NotImplementedError()
+
+    @property
+    @abc.abstractmethod
+    def ipython_serializer(self) -> IPythonSerializer[t.Any]:
+        """Return IPythonSerializer instance."""
+        raise NotImplementedError()
+
+    @property
+    @abc.abstractmethod
+    def html_serializer(self) -> HtmlSerializer[t.Any]:
+        """Return HtmlSerializer instance."""
+        raise NotImplementedError()
+
+    def show(
+        self,
+        as_widget: bool = True,
+        unique_id: t.Optional[str] = None,
+        **kwargs
+    ) -> t.Optional[HTMLFormatter]:
+        """Display result.
+
+        Parameters
+        ----------
+        as_widget : bool, default True
+            whether to display result with help of ipywidgets or not
+        unique_id : Optional[str], default None
+            unique identifier of the result output
+        **kwrgs :
+            other key-value arguments will be passed to the `Serializer.serialize`
+            method
+
+        Returns
+        -------
+        Optional[HTMLFormatter] :
+            when used by sphinx-gallery
+        """
+        if 'sphinx_gallery' in pio.renderers.default:
+            # TODO: why we need this? add comments
+            html = widget_to_html_string(  # pylint: disable=redefined-outer-name
+                self.widget_serializer.serialize(output_id=unique_id, **kwargs),
+                title=get_result_name(self),
+                requirejs=True
+            )
+
+            class TempSphinx:
+                def _repr_html_(self):
+                    return html
+
+            return TempSphinx()
+
+        if is_colab_env() and as_widget is True:
+            widget = self.widget_serializer.serialize(**kwargs)
+            content = widget_to_html_string(widget, title=get_result_name(self))
+            display_html(content, raw=True)
+        elif is_colab_env() and as_widget is False:
+            display(*self.ipython_serializer.serialize(**kwargs))
+        elif as_widget is True:
+            widget = self.widget_serializer.serialize(output_id=unique_id, is_for_iframe_with_srcdoc=True, **kwargs)
+            content = widget_to_html_string(widget, title=get_result_name(self))
+            content = iframe(srcdoc=content)
+            display_html(content, raw=True)
+        else:
+            # TODO:
+            # output of a HTML Serializers looks to plain
+            # it need some style
+            display_html(
+                iframe(srcdoc=self.html_serializer.serialize(
+                    output_id=unique_id,
+                    full_html=True,
+                    include_requirejs=True,
+                    include_plotlyjs=True,
+                    is_for_iframe_with_srcdoc=True,
+                    **kwargs
+                )),
+                raw=True
+            )
+
+    # def show_in_iframe(
```

Comment:
why is that in comment?

Reply:
I thought that we going to use iframe by default, therefore, commented it

uncommented (bring it back)